src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -363,6 +374,11 @@ ErrorMessageOr<std::unique_ptr<InstrumentedProcess>> InstrumentedProcess::Create
   process->injected_library_path_ = std::filesystem::canonical(library_path);
   ORBIT_LOG("Injecting library \"%s\" into process %d", process->injected_library_path_, pid);
 
+  // If we already injected the library in a previous run of OrbitService we merely need to retrieve
+  // some function pointers and create a new return trampoline so we can exit early below (skipping
+  // the initialization of the library that has already been done).
+  OUTCOME_TRY(const bool already_injected, AlreadyInjected(pid));
+
   auto library_handle_or_error = DlopenInTracee(pid, library_path, RTLD_NOW);
```

Comment:
I would add a comment here explaining that `DlopenInTracee` (and `dlopen` itself) doesn't cause the library to be injected a second time (which is not obvious as it was one of the questions we had in the bug).

Reply:
Rephrased the entire comment and added a bit about dlopen.